src/main.rs

```diff
@@ -121,13 +121,17 @@ fn snapshot(pid: pid_t) -> Result<(), Error> {
     Ok(())
 }
 
-fn record(output_filename: &Path, pid: pid_t, sample_rate: u32, is_subcommand: bool) -> Result<(), Error> {
+fn record(output_filename: &Path, pid: pid_t, sample_rate: u32, duration: u64, is_subcommand: bool) -> Result<(), Error> {
```

Comment:
it would be more idiomatic Rust for duration to be an `Option<u64>`, so that duration = None means that duration is unset instead of duration = 0

Reply:
it could actually even be an `Option<Duration>`!

Reply:
Sweet, will update.